components/list/tests/unit/list.spec.tsx

```diff
@@ -0,0 +1,168 @@
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { List, Item } from '@westpac/list';
+import { AppleIcon } from '@westpac/icon';
+import wbc from '@westpac/wbc';
+import { blenderList } from '../../src/overrides/list';
+import { blenderIcon } from '../../src/overrides/icon';
+import { overridesTest } from '../../../../helpers/tests/overrides-test';
+import { nestingTest } from '../../../../helpers/tests/nesting-test';
+import { ListProps } from '../../src/List';
+
+const SimpleList = (props: ListProps) => (
+	<List {...props}>
+		<Item>item1</Item>
+		<Item>item2</Item>
+		<Item>item3</Item>
+	</List>
+);
+
+const SimpleListWithIcon = (props: ListProps) => (
+	<List
+		type="icon"
+		icon={() => {
+			<AppleIcon copyrightYear="2020" />;
+		}}
+		{...props}
+	>
+		<Item>item1</Item>
+		<Item>item2</Item>
+		<Item>item3</Item>
+	</List>
+);
+
+overridesTest({
+	name: 'list',
+	overrides: ['List', 'Item', 'Content'],
+	Component: (props: ListProps) => <SimpleList {...props} />,
+});
+
+overridesTest({
+	name: 'list',
+	overrides: ['Icon'],
+	Component: (props: ListProps) => <SimpleListWithIcon {...props} />,
+});
+
+nestingTest({
+	name: 'list',
+	Component: (props: ListProps) => <SimpleList {...props} />,
+});
+
+describe('List component', () => {
+	const WBCList = (props: ListProps) => (
+		<GEL brand={wbc}>
+			<List {...props} />
+		</GEL>
+	);
+
+	it('renders List with default props', () => {
+		const { container } = render(<WBCList />);
+		expect(container).toBeInTheDocument();
+	});
+
+	it('renders List correctly', () => {
+		const defaultProps: ListProps = {
+			assistiveText: 'Test Assistive Text',
+			type: 'tick',
+			look: 'hero',
+			spacing: 'large',
+			data: [
+				<>
+					<Item>item1</Item>
+					<Item>item2</Item>
+					<Item>item3</Item>
+				</>,
+			],
+		};
+		const { getByText, getByLabelText } = render(<WBCList {...defaultProps} />);
+		expect(getByLabelText(/Test Assistive Text/i)).toBeInTheDocument();
+		expect(getByText(/item1/i)).toBeInTheDocument();
+		expect(getByText(/item2/i)).toBeInTheDocument();
+		expect(getByText(/item3/i)).toBeInTheDocument();
+	});
+
+	it('renders nested List correctly', () => {
+		const WBCNestedList = () => (
+			<GEL brand={wbc}>
+				<List
+					data={[
+						'item1',
+						{
+							text: 'item2',
```

Comment:
I just wonder where the "text" prop is in the Item?

Reply:
It is from data props. It would be easy to understand if you refer to data-driven files from the example directory.